libobs/obs-internal.h

```diff
@@ -520,6 +520,7 @@ typedef void (*obs_destroy_cb)(void *obj);
 
 struct obs_context_data {
 	char *name;
+	char *uuid;
```

Comment:
Should probably be a const char, so it can't be modified.

Reply:
Since this is allocated at runtime, making it const would mean you could not free the pointer later on. The context data of any OBS object must also never be accessed directly outside the internals, and `obs_source_get_uuid()` should be used instead.

Reply:
See https://github.com/obsproject/obs-studio/blob/fea4a19c8b146d01a0357ba59a1e6003e7018264/libobs/obs-source.h#L223 and https://github.com/obsproject/obs-studio/blob/fea4a19c8b146d01a0357ba59a1e6003e7018264/libobs/obs-source.c#L743 as an example of how it would be done.

Reply:
I'm surprised the compiler doesn't complain about the recasting of a const variable to a non-const one here.